snuba/web/rpc/storage_routing/routing_strategies/outcomes_based.py

```diff
@@ -153,15 +153,18 @@ def _get_routing_decision(self, routing_context: RoutingContext) -> RoutingDecis
             span.set_data(
                 "downsampling_mode",
                 "highest_accuracy"
-                if self._is_highest_accuracy_mode(routing_context)
+                if self._is_highest_accuracy_mode(routing_context.in_msg)
```

Comment:
This is not the `in_msg.meta` so I think it will fail.

Fixing the typing to return `ResponseMeta` from `extract_message_meta` and only accepts `ResponseMeta` as a parameter type in `_is_highest_accuracy_mode` would help detect this.